libs/sdk-core/src/persist/migrations.rs

```diff
@@ -460,5 +460,99 @@ pub(crate) fn current_sync_migrations() -> Vec<&'static str> {
+         private_key BLOB NOT NULL UNIQUE,
+         public_key BLOB NOT NULL UNIQUE,
+         swapper_public_key BLOB NOT NULL UNIQUE,
+         script BLOB NOT NULL UNIQUE,
+         min_allowed_deposit INTEGER NOT NULL,
+         max_allowed_deposit INTEGER NOT NULL
+        ) STRICT;
+
+        CREATE TABLE IF NOT EXISTS swap_refunds (
+         bitcoin_address TEXT NOT NULL,
+         refund_tx_id TEXT NOT NULL,
+         PRIMARY KEY (bitcoin_address, refund_tx_id)
+        ) STRICT;
+
+        CREATE TABLE IF NOT EXISTS payments_external_info (
+         payment_id TEXT NOT NULL PRIMARY KEY,
+         lnurl_success_action TEXT,
+         ln_address TEXT,
+         lnurl_metadata TEXT
+        ) STRICT;
+
+        CREATE TABLE IF NOT EXISTS reverse_swaps (
+         id TEXT PRIMARY KEY NOT NULL,
+         created_at_block_height INTEGER NOT NULL,
+         preimage BLOB NOT NULL UNIQUE,
+         private_key BLOB NOT NULL UNIQUE,
+         claim_pubkey TEXT NOT NULL,
+         timeout_block_height INTEGER NOT NULL,
+         invoice TEXT NOT NULL UNIQUE,
+         onchain_amount_sat INTEGER NOT NULL,
+         sat_per_vbyte INTEGER NOT NULL,
+         redeem_script TEXT NOT NULL
+        ) STRICT;
+
+        CREATE TABLE IF NOT EXISTS sync_requests (
+         id INTEGER PRIMARY KEY AUTOINCREMENT,
+         changed_table TEXT NOT NULL
+        ) STRICT;
+
+        CREATE TRIGGER IF NOT EXISTS sync_requests_swaps
+         AFTER INSERT ON swaps
+        BEGIN
+         INSERT INTO sync_requests(changed_table) VALUES('swaps');
+        END;
+
+        CREATE TRIGGER IF NOT EXISTS sync_requests_swap_refunds
+         AFTER INSERT ON swap_refunds
+        BEGIN
+         INSERT INTO sync_requests(changed_table) VALUES('swap_refunds');
+        END;
+
+        CREATE TRIGGER IF NOT EXISTS sync_requests_reverse_swaps
+         AFTER INSERT ON reverse_swaps
+        BEGIN
+         INSERT INTO sync_requests(changed_table) VALUES('reverse_swaps');
+        END;
+
+        CREATE TRIGGER IF NOT EXISTS sync_requests_payments_external_info
+         AFTER INSERT ON payments_external_info
+        BEGIN
+         INSERT INTO sync_requests(changed_table) VALUES('payments_external_info');
+        END;
+        ",
+        "
+        ALTER TABLE payments_external_info RENAME TO payments_external_info_old;
+
+        CREATE TABLE payments_external_info (
+         payment_id TEXT NOT NULL PRIMARY KEY,
+         lnurl_success_action TEXT,
+         ln_address TEXT,
+         lnurl_metadata TEXT,
+         lnurl_withdraw_endpoint TEXT
+        ) STRICT;
+
+        INSERT INTO payments_external_info
+         (payment_id, lnurl_success_action, ln_address, lnurl_metadata, lnurl_withdraw_endpoint)
+         SELECT
+          payment_id,
+          lnurl_success_action,
+          ln_address,
+          lnurl_metadata,
+          NULL
+         FROM payments_external_info_old;
+
+        DROP TABLE payments_external_info_old;
+        ",
```

Comment:
There are triggers on this table. I think triggers reference the table id, not the table name. So renaming the table, then creating a new one with the same name will malfunction the triggers.

Reply:
Thinking about it, recreating the trigger here will fail the main db migration, because that's also unconditionally creating the trigger.

Reply:
@JssDWt @ok300  Perhaps now that we have added the triggers here we can replace the main db triggers with empty statements?
And we can also re-add the trigger after the table deletion (because sqlite implicitly deletes all triggers associated with the table before the table is deleted)?

Reply:
I think we should just let the main db migrations run first, and the sync migrations second. Then you also don't need to recreate the table here.

So do:
`CREATE TABLE IF NOT EXISTS ...` on the `payments_external_info` table.
then add the column.

Reply:
I see the main table is actually created first, that was my confusion.

So just the CREATE TABLE IF NOT EXISTS ... on the payments_external_info table.
then add the column, that should be enough.

Reply:
I was wrong about the order of migrations, I thought the sync migration ran first, and then the main migration, but it's the other way around. So just readding the trigger is enough. No need to change the main migrations.

Reply:
I am thinking even if the main db run first it is confusing to try and foresee all the combinations with the main db and sync db so we can take advantage of this refactoring to strictly isolate the migrations, each on its own file and schema and also remove any future confusion. That's why I suggested to nullify these statements.

Reply:
I moved the `payments_external_info` trigger right after the re-creation of the table, so the trigger is created on the new table: ccb7c850f9f5c538fd08695a4fa9b4011f7acce0

Reply:
@roeierez if I understood your point correctly to nullify main migration statments which are not needed anymore, this removes the triggers and the `sync_requests` table from the main migrations: 1eab2935fc3e28e5bd21958fca50ba2ff6d5fda7

They all of now exist in the sync migration below.